src/crimson/os/seastore/cache.h

```diff
@@ -435,24 +332,27 @@ class Cache {
   }
 
   /**
-   * get_extent
+   * get_caching_extent
    *
    * returns ref to extent at offset~length of type T either from
    * - t if modified by t
    * - extent_set if already in cache
    * - disk
    *
    * t *must not* have retired offset
+   *
+   * Note, the current implementation leverages parent-child
+   * pointers in LBA instead, so it should only be called in tests.
```

Comment:
Not in this context, but I've think since we retrieve all extents through lba/backref trees, `Cache::extents` only serves as an extent holder instead of an extent index and may not be necessary of type `ExtentIndex`, it seems an intrusive list is enough, What do you think? This should avoid the unnecessary cpu overhead caused by inserting extent into `Cache::extents`.

Reply:
> Not in this context.

Is it fine to keep this comment, or you suggest to modify in some way?

> But I've thinking that, since we retrieve all extents through lba/backref trees, Cache::extents only serves as an extent holder instead of an extent index and may not be necessary of type ExtentIndex, it seems an intrusive list is enough, What do you think? This should avoid the unnecessary cpu overhead caused by inserting extent into Cache::extents.

This makes sense to me. Also refer to https://github.com/ceph/ceph/pull/47749#discussion_r1162507311 that I thought we should skip `query_cache()` in the non-test path.

Reply:
> > Not in this context.
> 
> Is it fine to keep this comment, or you suggest to modify in some way?
> 

Ah, sorry, I meant what I was going to suggest was not in this context, not the comment in the code:-)